integration/compatibility_test.py

```
@@ -232,35 +246,35 @@ def compare_row(l, r, key_type):
-            assert isinstance(r[rks[i]], list)
-            if len(l[lks[i]]) != len(r[rks[i]]):
-                print("mismatch vector field length: ", l[lks[i]], " ", r[rks[i]])
+            assert isinstance(l_filtered[lks[i]], list)
+            assert isinstance(r_filtered[rks[i]], list)
+            if len(l_filtered[lks[i]]) != len(r_filtered[rks[i]]):
+                print("mismatch vector field length: ", l_filtered[lks[i]], " ", r_filtered[rks[i]])
                 return False
-            for i in range(l[lks[i]]):
-                if not compare_number_eq(l[lks[i]][i], r[rks[i]][i]):
-                    print("mismatch vector field value: ", l[lks[i]], " ", r[rks[i]])
+            for i in range(l_filtered[lks[i]]):
```

Ruff (0.15.14)

[warning] Loop control variable `i` overrides iterable it iterates

(B020)